src/data_chain/chain.rs

```diff
@@ -185,3 +190,132 @@ impl Chain {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use data_chain::{Block, NodeState, State};
+    use full_info::FullInfo;
+    use maidsafe_utilities::SeededRng;
+    use routing_table::Prefix;
+    use rust_sodium;
+    use rust_sodium::crypto::sign;
+    use tiny_keccak::sha3_256;
+    use xor_name::XorName;
+
+    #[test]
+    fn is_valid_pair() {
+        let mut rng = SeededRng::thread_rng();
+        unwrap!(rust_sodium::init_with_rng(&mut rng));
+
+        let group_size = 8;
+        let chain = Chain::from_blocks(Vec::new(), group_size);
+
+        let full_info = FullInfo::node_new(0);
+
+        let (pub_key_0, _) = sign::gen_keypair();
+        let name = XorName(sha3_256(&pub_key_0.0));
+        let prefix = Prefix::new(1, name, 0);
+
+        let node_state = NodeState {
+            state: State::ElderLive,
+            public_key: pub_key_0,
+            age: 5,
+            section: prefix,
+        };
+
+        // Check valid pairs of blocks.
+        let valid_pairs = [
```

Comment:
I think these are the wrong way round.  It should be Offline or Demoted or Relocated followed by Live.  Maybe even best to leave the SplitTo type out just now until we decide how splits are actually to be handled?